src/main/java/org/zephyrsoft/sdb2/model/Song.java

```diff
@@ -222,27 +271,130 @@ public boolean equals(Object obj) {
 		if (getClass() != obj.getClass())
 			return false;
 		Song other = (Song) obj;
-		if (chordSequence == null) {
-			if (other.chordSequence != null)
-				return false;
-		} else if (!chordSequence.equals(other.chordSequence))
-			return false;
-		if (lyrics == null) {
-			if (other.lyrics != null)
-				return false;
-		} else if (!lyrics.equals(other.lyrics))
-			return false;
-		if (title == null) {
-			if (other.title != null)
+		return equalsAllowNull(uuid, other.uuid) &&
+			equalsAllowNull(title, other.title) &&
+			equalsAllowNull(composer, other.composer) &&
+			equalsAllowNull(authorText, other.authorText) &&
+			equalsAllowNull(authorTranslation, other.authorTranslation) &&
+			equalsAllowNull(publisher, other.publisher) &&
+			equalsAllowNull(additionalCopyrightNotes, other.additionalCopyrightNotes) &&
+			equalsAllowNull(language, other.language) &&
+			equalsAllowNull(songNotes, other.songNotes) &&
+			equalsAllowNull(tonality, other.tonality) &&
+			equalsAllowNull(chordSequence, other.chordSequence) &&
+			equalsAllowNull(drumNotes, other.drumNotes) &&
+			equalsAllowNull(tempo, other.tempo) &&
+			equalsAllowNull(lyrics, other.lyrics);
+	}
+
+	@Override
+	public String toString() {
+		return "SONG[" + title + "|" + uuid + "]";
+	}
+
+	private static boolean equalsAllowNull(String str, String str2) {
+		if (str == null || str.isEmpty()) {
+			if (str2 != null && !str2.isEmpty())
 				return false;
-		} else if (!title.equals(other.title))
+		} else if (!str.equals(str2))
```

Comment:
Please always use curly braces for blocks (even if the code didn't contain them in the first place).

Reply:
ok.